hq/app/schedule/IamDisableAccessKeys.scala

```diff
@@ -0,0 +1,53 @@
+package schedule
+
+import aws.AwsAsyncHandler.{awsToScala, handleAWSErrs}
+import aws.iam.IAMClient.SOLE_REGION
+import aws.{AwsClient, AwsClients}
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementAsync
+import com.amazonaws.services.identitymanagement.model.{UpdateAccessKeyRequest, UpdateAccessKeyResult}
+import model._
+import play.api.Logging
+import schedule.IamFlaggedUsers.{hasOutdatedHumanKey, hasOutdatedMachineKey}
+import schedule.IamListAccessKeys.listAccountAccessKeys
+import utils.attempt.Attempt
+
+import scala.concurrent.ExecutionContext
+
+object IamDisableAccessKeys extends Logging {
+
+  def disableAccessKeys(account: AwsAccount, vulnerableUsers: Seq[VulnerableUser], iamClients: AwsClients[AmazonIdentityManagementAsync])
+    (implicit ec: ExecutionContext): Unit = {
+    // this does the work of taking our vulnerable users who have been flagged as potentially needing their access keys disabled
+    // and converts that vulnerableUser into a user that has it's access key id attached to it
+    val vulnerableUserWithAccessKeyId: Attempt[List[VulnerableUserWithAccessKeyId]] = listAccountAccessKeys(account, vulnerableUsers, iamClients)
+    vulnerableUserWithAccessKeyId.map { users =>
+      findAccessKeysToDisable(users).map { user =>
+        val key = user.accessKey
+        logger.info(s"attempting to disable access key id ${key.id}.")
+        for {
+          client <- iamClients.get(account, SOLE_REGION)
+          updateAccessKeyResult <- disableAccessKey(key, client)
+        } yield {
```

Comment:
I couldn't get this to compile when I removed the yield! You'll have to show me how to do this!

Reply:
This change is superseded by this one: https://github.com/guardian/security-hq/pull/283#discussion_r679301060. So will ignore this in favour of the other.

Reply:
Yeah that's true, but I saw your earlier comment about it not compiling and dug into that - it's because without the yield a for-comprehension will use `foreach` under the hood as opposed to `map` / `flatMap`, which makes sense because we're not generating a new collection, just performing a side effect. And that leads to a compilation error because `foreach` hasn't been defined on `Attempt` yet